RandLAPACK/misc/rl_linops.hh

```diff
@@ -0,0 +1,79 @@
+#ifndef randlapack_linops_h
+#define randlapack_linops_h
+
+#include "rl_blaspp.hh"
+#include "rl_lapackpp.hh"
+#include "rl_util.hh"
+
+#include <RandBLAS.hh>
+#include <iostream>
+#include <cmath>
+#include <algorithm>
+#include <vector>
+#include <cstdint>
+
+namespace RandLAPACK {
+
+template <typename T>
+struct SymmetricLinearOperator {
+
+    const int64_t m;
+
+    SymmetricLinearOperator(int64_t m) : m(m) {};
+
+    /* The semantics of this function are similar to blas::symm.
+        * We compute
+        *      C = alpha * A * B + beta * C
+        * where this SymmetricLinearOperator object represents "A"
+        * and "B" has "n" columns.
+    */
+    virtual void operator()(
+        blas::Layout layout,
```

Comment:
This defines the layout of A, B, C?

Reply:
Only of ``B, C``. Since we're working with a linear operator it's artificial to impose a layout specification for ``A``.

Reply:
I've updated the documentation here. I've also updated the implementation of ExplicitSymLinOp's ``operator()`` function so that it can reconcile differences in the layout of its internal buffer and the layout of ``(B, C)`` that might be passed when the ExplicitSymLinOp is called.